### PR TITLE
🏗 Configure CircleCI to email build/release-on-duty on main/release branch failures, respectively

### LINKED ISSUE #32730: Reinstate push build failure notifications for build cop
With TravisCI the build cops got an email notification every time a push build failed. We don't seem to get them from CircleCI.

### PR DESCRIPTION
**Pull request by danielrozenberg**
Fixes #32730

### CODE REVIEW

**Comment by danielrozenberg** [comment]
@rsimha let's talk on monday about this PR :)

File: .circleci/fail_fast.sh [hunk 1]
```diff
@@ -25,8 +25,13 @@ YELLOW() { echo -e "\n\033[0;33m$1\033[0m"; }
 # For push builds, continue in spite of failures so that other jobs like
 # bundle-size and visual-diff can establish their baselines for this commit.
 # Without this, our custom bots will not be able to function correctly.
-if [[ "$CIRCLE_BRANCH" == "master" || "$CIRCLE_BRANCH" =~ ^amp-release-* ]]; then
-  echo $(YELLOW "Not canceling build in spite of failures because $CIRCLE_BRANCH is not a PR branch.")
+if [[ "$CIRCLE_BRANCH" == "master" ]]; then
+  echo $(YELLOW "This master branch build failed, notifying @ampproject/build-on-duty.")
+  echo $(RED "NotImplementedError")
+  exit 0
+if [[ "$CIRCLE_BRANCH" =~ ^amp-release-* ]]; then
```
Comment (rsimha) [inline .circleci/fail_fast.sh:32]: nit: This should be `elif`

**Review by rsimha — COMMENTED** [review]
Interesting idea! I did some initial explorations of native notifications (https://circleci.com/docs/2.0/notifications), but they don't provide the granularity or flexibility that this approach could achieve.

Let's discuss offline.

File: .circleci/fail_fast.sh [hunk 2]
```diff
@@ -22,11 +22,24 @@ set -e
 RED() { echo -e "\n\033[0;31m$1\033[0m"; }
```
Comment (rsimha) [inline .circleci/fail_fast.sh:22]: Once this works, update the top-level comment on line 17 to mention the emailing. And maybe add  a reference link for the sendgrid format too?

**Review by rsimha — APPROVED** [review]
Looks promising! Pre-emptive approval to unblock you.

File: .circleci/fail_fast.sh (hunk 2 above)
Comment (danielrozenberg) [reply .circleci/fail_fast.sh:22]: Done.

File: .circleci/fail_fast.sh (hunk 1 above)
Comment (danielrozenberg) [reply .circleci/fail_fast.sh]: Done.

File: .circleci/fail_fast.sh [hunk 3]
```diff
@@ -16,17 +16,52 @@
...
+          }
+        ]
+      }'
+}
+
 # For push builds, continue in spite of failures so that other jobs like
 # bundle-size and visual-diff can establish their baselines for this commit.
 # Without this, our custom bots will not be able to function correctly.
-if [[ "$CIRCLE_BRANCH" == "main" || "$CIRCLE_BRANCH" =~ ^amp-release-* ]]; then
-  echo $(YELLOW "Not canceling build in spite of failures because $CIRCLE_BRANCH is not a PR branch.")
```
Comment (rsimha) [inline .circleci/fail_fast.sh:29]: This logging message and the comment on lines 25-27 are related to each other, and are meant to indicate why fast-fail isn't being applied to certain push builds.

Can we reinstate this message in both cases where we're not fast-failing?

Comment (danielrozenberg) [reply .circleci/fail_fast.sh:29]: Done

**Review by rsimha — APPROVED** [review]
LGTM! Glad this is in place. Let's see how things shake out.

**Pull request merged**